tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/llk_math_eltwise_unary_sfpu_macros.h

```diff
@@ -0,0 +1,176 @@
+
+// For ops where compute takes extra args
+#define SFPU_UNARY_PARAMS_KERNEL_EXTRA_ARGS(FN, MODE, APPROXIMATE, DST_IDX, PARAM0, PARAM1) \
+    _llk_math_eltwise_unary_sfpu_params_<APPROXIMATE>(                                      \
+        ckernel::sfpu::FN<APPROXIMATE>, DST_IDX, (int)VectorMode::MODE, PARAM0, PARAM1);
+
+// For ops with multiple template parameters and one runtime parameter (e.g., scale)
+#define SFPU_TEMPLATE_PARAMS_KERNEL(                                                                                  \
+    OP,                                                                                                               \
+    APPROXIMATE,                                                                                                      \
+    FAST_APPROX,                                                                                                      \
+    SCALE_EN,                                                                                                         \
+    SKIP_POSITIVE_CHECK,                                                                                              \
+    ITERATIONS,                                                                                                       \
+    IS_FP32_DEST_ACC_EN,                                                                                              \
+    DST_IDX,                                                                                                          \
+    VECTOR_MODE,                                                                                                      \
+    SCALE)                                                                                                            \
+    _llk_math_eltwise_unary_sfpu_params_<APPROXIMATE>(                                                                \
+        ckernel::sfpu::                                                                                               \
+            calculate_##OP<APPROXIMATE, FAST_APPROX, SCALE_EN, ITERATIONS, SKIP_POSITIVE_CHECK, IS_FP32_DEST_ACC_EN>, \
+        DST_IDX,                                                                                                      \
+        VECTOR_MODE,                                                                                                  \
+        SCALE)
+
+// For kernels with one template parameter and one extra runtime argument.
+#define SFPU_UNARY_ONE_PARAM_KERNEL_FN(FN, MODE, APPROXIMATE, DST_IDX, PARAM0) \
+    _llk_math_eltwise_unary_sfpu_params_<APPROXIMATE>(                         \
+        ckernel::sfpu::FN<APPROXIMATE>, DST_IDX, (int)VectorMode::MODE, PARAM0)
+
+#define SFPU_UNARY_ONE_PARAM_KERNEL_FN_FLOAT(FN, MODE, APPROXIMATE, DST_IDX, PARAM0) \
+    _llk_math_eltwise_unary_sfpu_params_<APPROXIMATE>(                               \
+        ckernel::sfpu::_##FN##_<sfpi::vFloat, APPROXIMATE, 8, uint32_t>, DST_IDX, (int)VectorMode::MODE, PARAM0)
```

Comment:
Seeing as the kernel accepts `int` and `VectorMode` is an unscoped enum of underlying type `int`, this cast is unnecessary: https://godbolt.org/z/f3dqdPs5G

And again, for improved code navigation let's avoid concatenating `_` to the parameter `FN` passed in.